components/containers/core/browser/prefs.cc

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/containers/core/browser/prefs.h"
+
+#include <utility>
+
+#include "brave/components/containers/core/browser/pref_names.h"
+#include "brave/components/containers/core/mojom/containers.mojom.h"
+#include "components/pref_registry/pref_registry_syncable.h"
+#include "components/prefs/pref_service.h"
+
+namespace containers {
+
+void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
+  registry->RegisterListPref(prefs::kContainersList,
+                             user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
+}
+
+std::vector<mojom::ContainerPtr> GetContainersFromPrefs(
+    const PrefService& prefs) {
+  std::vector<mojom::ContainerPtr> containers;
```

Comment:
per above comment on feature flag guard, should we CHECK here that the feature is enabled?